src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -412,44 +439,71 @@ def get_yaml_and_values(cmd_value, timeout, file):
     return yaml_template, values_content
 
 
-def get_custom_registry_credentials(cmd, auth_mode=None, login_server=None, username=None, password=None):
+def get_custom_registry_credentials(cmd,
+                                    auth_mode=None,
+                                    login_server=None,
+                                    username=None,
+                                    password=None,
+                                    kv_username=None,
+                                    kv_password=None,
+                                    identity=None):
     """Get the credential object from the input
     :param str auth_mode: The login mode for the source registry
     :param str login_server: The login server of custom registry
     :param str username: The username for custom registry
     :param str password: The password for custom registry
+    :param str kv_username: The key vault secret URI of the username to login to the custom registry
+    :param str kv_password: The key vault secret URI of the password to login to the custom registry
+    :param str identity: The task managed identity used for the credential
     """
+
     source_registry_credentials = None
     if auth_mode:
         SourceRegistryCredentials = cmd.get_models('SourceRegistryCredentials')
-        source_registry_credentials = SourceRegistryCredentials(login_mode=auth_mode)
+        source_registry_credentials = SourceRegistryCredentials(
+            login_mode=auth_mode)
 
+    is_remove = False
     custom_registries = None
     if login_server:
+        # if null username and password (or identity), then remove the credential
+        custom_reg_credential = None
+        # Validate if username or password are used, exactly one of each type exist is given
+        if username is not None and kv_username is not None:
```

Comment:
`if username and kv_username:`? I remember the same question, how would you handle empty vs. null `username`/`kv_username`?

Reply:
See logic simplification. Both empty and null or blocked because python expects one argument.